File: java-reactive-logger/src/main/java/hu/numichi/reactive/logger/MDCContextWriter.java
package hu.numichi.reactive.logger;

import reactor.util.annotation.NonNull;
import reactor.util.context.Context;

import java.util.Map;

public final class MDCContextWriter {
    private MDCContextWriter() {
    }
    
    @NonNull
    public static Context put(
        @NonNull Context context,
        @NonNull Map<String, String> mdc
    ) {
        return context.put(ReactiveLogger.DEFAULT_REACTOR_CONTEXT_MDC_KEY, mdc);
    }
    
    @NonNull
    public static Context put(
        @NonNull Context context,
        @NonNull String mdcContextKey,
        @NonNull Map<String, String> mdc
    ) {
        return context.put(mdcContextKey, mdc);
    }
    
    @NonNull
    public static Context put(
        @NonNull Context context,
        @NonNull MDC mdc
    ) {
        return context.put(mdc.getMdcContextKey(), mdc.getMdcMap());
    }
}

File: kotlin-coroutine-reactive-logger/src/main/kotlin/hu/numichi/kotlin/reactive/logger/MDCContextWriter.kt
package hu.numichi.kotlin.reactive.logger;

import hu.numichi.reactive.logger.ReactiveLogger;
import reactor.util.context.Context;

object MDCContextWriter {
    fun put(context: Context, mdc: Map<String, String>): Context {
        return context.put(ReactiveLogger.DEFAULT_REACTOR_CONTEXT_MDC_KEY, mdc);
    }

    fun put(context: Context, mdcContextKey: String, mdc: Map<String, String>): Context {
        return context.put(ReactiveLogger.DEFAULT_REACTOR_CONTEXT_MDC_KEY, mdc);
    }
}
